backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/slider/internal/BpkRangeSliderImpl.kt

```diff
@@ -0,0 +1,217 @@
+    modifier: Modifier = Modifier,
+    lowerThumbLabel: String? = null,
+    upperThumbLabel: String? = null,
+) {
+    val startInteractionSource = remember { MutableInteractionSource() }
+    val endInteractionSource = remember { MutableInteractionSource() }
+    RangeSlider(
+        value = value,
+        onValueChange = onValueChange,
+        modifier = modifier,
+        enabled = enabled,
+        startInteractionSource = startInteractionSource,
+        endInteractionSource = endInteractionSource,
+        valueRange = minValue..maxValue,
+        steps = steps,
+        onValueChangeFinished = onValueChangeFinished,
+        startThumb = {
+            lowerThumbLabel?.let {
+                SlideRangeLabel(
+                    label = it,
+                    enabled = enabled,
+                    interactionSource = startInteractionSource,
+                    value = value.start,
+                    valueRange = minValue..maxValue,
+                    steps = steps,
+                    sliderRangeType = stringResource(id = R.string.range_start),
+                    onValueChange = { changedVal -> onValueChange(changedVal..value.endInclusive) },
+                    onValueChangeFinished = onValueChangeFinished,
+                )
+            } ?: SliderDefaults.Thumb(
+                interactionSource = startInteractionSource,
+                colors = sliderColors(),
+                enabled = enabled,
+            )
+        },
+        endThumb = {
+            upperThumbLabel?.let {
+                SlideRangeLabel(
+                    label = it,
+                    enabled = enabled,
+                    interactionSource = endInteractionSource,
+                    value = value.endInclusive,
+                    valueRange = minValue..maxValue,
+                    steps = steps,
+                    sliderRangeType = stringResource(id = R.string.range_end),
+                    onValueChange = { changedVal -> onValueChange(value.start..changedVal) },
+                    onValueChangeFinished = onValueChangeFinished,
+                )
+            } ?: SliderDefaults.Thumb(
+                interactionSource = endInteractionSource,
+                colors = sliderColors(),
+                enabled = enabled,
+            )
+        },
+        colors = sliderColors(),
+    )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun SlideRangeLabel(
+    value: Float,
+    enabled: Boolean,
+    label: String,
+    onValueChange: (Float) -> Unit,
+    sliderRangeType: String,
+    interactionSource: MutableInteractionSource,
+    valueRange: ClosedFloatingPointRange<Float>,
+    modifier: Modifier = Modifier,
+    onValueChangeFinished: (() -> Unit)? = null,
+    steps: Int = 0,
+) {
+    Column(
+        modifier = modifier
+            .sliderSemantics(
+                value = value,
+                enabled = enabled,
+                description = "$sliderRangeType, $label",
```

Comment:
I think rather than combining the two it'd be better the `sliderRangeType` to be used as the `contentDescription` rather than the `stateDescription` - that when updating the range you'll only hear the state update, rather than the content description again